### ISSUE TITLE
Question about the calculation of time-gap matrix

### ISSUE THREAD

**Issue by GMavrak** [WenjieDu/PyPOTS#294]
Hello,
I wish you a happy new year.

I have a question about the calculation of the delta matrix. By examining the nested function of the [ _parse_delta_torch](https://github.com/WenjieDu/PyPOTS/blob/main/pypots/data/utils.py) function, the following code for populating the matrix for time-step > 1:
`
d.append(torch.ones(1, n_features, device=device) + (1 - mask[step]) * d[-1])
`
Shouldn't be mask[step - 1] instead of mask[step], as the calculation of the time-gap in time-step t depends upon the mask value of the time-step t-1?

Thank you,
George

**Comment by WenjieDu** [comment]
@GMavrak Thank you! I'll investigate this.

**Comment by GMavrak** [comment]
@WenjieDu I have reviewed the PR and confirm that it is OK. 

Concerning the comments, I see the point. However, the way that the time-gap matrix is defined, as far as I understand it, is:
- not dependent on the features, i.e. it does not make sense to compare the features in this context.
- each element in the delta matrix is the time-gap from the last (valid) observation to the current timestamp, irrespective of the timestamp's value.

Thank you,
George
